src/pages/home/report/ReportActionsList.js

```diff
@@ -145,6 +145,10 @@ class ReportActionsList extends React.Component {
     }
 
     render() {
+        // Non-animating skeleton UI represents that chat history exists but is not loaded locally in Onyx.
+        // While offline if there is at least one non-pending action or if this is a new chat then we don't need to show the non-animating skeleton UI.
+        const shouldShowNonAnimatingSkeleton = !(_.some(this.props.sortedReportActions, action => !action.pendingAction || action.actionName === CONST.REPORT.ACTIONS.TYPE.CREATED));
```

Comment:
Parens are unneccessary

Reply:
How is this different from `this.props.report.isLoadingReportActions`?

Reply:
`shouldShowNonAnimatingSkeleton` is not a great variable name either. It should ideally reflect what it represents - so in this case ... whether or not we still have report actions to load? (Hence why I have my question above)

Reply:
`this.props.isLoadingReportActions` should be `false` when we are offline(more [here](https://github.com/Expensify/App/pull/13233#discussion_r1037488570)), so I have updated some parts of code, so we don't call `openReport` when we are offline.

So we need another way to know whether we want to show the non-animating skeleton UI. (i.e., when we don't have any chat history in onyx and all we are left with are optimistic actions that were added by the user while being offline). So we will show the non-animating skeleton UI only if every action is optimistic and it is not a new chat report.

Reply:
I have updated the condition to be more clearer.
```js
// Non-animating skeleton UI represents that chat history exists but is not loaded locally in Onyx.
// While offline if there is at least one non-pending action or if this is a new chat then we don't need to show the non-animating skeleton UI.
const shouldShowNonAnimatingSkeleton = !this.props.report.isOptimisticReport && _.every(this.props.sortedReportActions, action => action.pendingAction);
```